File: utilities/src/main/java/com/adsamcik/signals/utilities/enums/ResolvedActivity.java
package com.adsamcik.signals.utilities.enums;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import static com.adsamcik.signals.utilities.enums.ResolvedActivity.IN_VEHICLE;
import static com.adsamcik.signals.utilities.enums.ResolvedActivity.ON_FOOT;
import static com.adsamcik.signals.utilities.enums.ResolvedActivity.STILL;
import static com.adsamcik.signals.utilities.enums.ResolvedActivity.UNKNOWN;

@IntDef({STILL, ON_FOOT, IN_VEHICLE, UNKNOWN})
@Retention(RetentionPolicy.SOURCE)
public @interface ResolvedActivity {
	int STILL = 0;
	int ON_FOOT = 1;
	int IN_VEHICLE = 2;
	int UNKNOWN = 3;
}

File: app/src/main/java/com/adsamcik/signalcollector/activity/ResolvedActivity.kt
package com.adsamcik.signalcollector.activity

import androidx.annotation.IntDef
import com.adsamcik.signalcollector.activity.ResolvedActivities.ResolvedActivity

/**
 * Singleton that contains [ResolvedActivity]. It is made this way so even the constants are contained within an object.
 */
object ResolvedActivities {
	const val STILL: Int = 0
	const val ON_FOOT: Int = 1
	const val IN_VEHICLE: Int = 2
	const val UNKNOWN: Int = 3

	@IntDef(STILL, ON_FOOT, IN_VEHICLE, UNKNOWN)
	@Retention(AnnotationRetention.SOURCE)
	annotation class ResolvedActivity
}
